pkg/client/presentproof/client.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package presentproof
+
+import (
+	"errors"
+
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/common/service"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/presentproof"
+)
+
+var (
+	errEmptyRequestPresentation = errors.New("request presentation message is empty")
+	errEmptyProposePresentation = errors.New("propose presentation message is empty")
+)
+
+// Provider contains dependencies for the protocol and is typically created by using aries.Context()
+type Provider interface {
+	Service(id string) (interface{}, error)
+}
+
+// ProtocolService defines the presentproof service.
+type ProtocolService interface {
+	service.DIDComm
+	Actions() ([]presentproof.Action, error)
+	ActionContinue(piID string, opt presentproof.Opt) error
+	ActionStop(piID string, err error) error
```

Comment:
nitpick: I would have placed these two methods on the `Action` objects themselves

Reply:
These functions are supposed to be used for async usage (rest API).  We need to continue with the protocol by `piID` (not by the action object). `ActionContinue` and `ActionStop` are using `service` under the hood. If I add these methods it means we will keep `service` in memory. It makes things complicated.